flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/FlinkContainerTestEnvironment.java

```diff
@@ -89,14 +87,16 @@ public void triggerTaskManagerFailover(JobClient jobClient, Runnable afterFailAc
         flinkContainer.restartTaskManager(
                 () -> {
                     try {
-                        CommonTestUtils.waitForJobStatus(
-                                jobClient,
-                                Arrays.asList(
-                                        JobStatus.FAILING, JobStatus.FAILED, JobStatus.RESTARTING),
-                                Deadline.fromNow(Duration.ofSeconds(30)));
+                        CommonTestUtils.waitForNoTaskRunning(
+                                () ->
+                                        flinkContainer
+                                                .getRestClusterClient()
+                                                .getJobDetails(jobClient.getJobID())
+                                                .get(),
+                                Deadline.fromNow(Duration.ofMinutes(1)));
                     } catch (Exception e) {
-                        throw new RuntimeException(
-                                "Error waiting for job entering failure status", e);
+                        throw new IllegalStateException(
+                                "Failed to wait for all task not running", e);
                     }
```

Comment:
I think this exception translation is not really necessary. Just change `public void restartTaskManager(Runnable afterFailAction)` to `RunnableWithException`
If you want to keep it, then please change the message to be a bit more helpful.